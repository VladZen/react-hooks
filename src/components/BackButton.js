import '@styles/components/back-button.scss'

import { Link } from 'react-router-dom'

import Icon from '@components/Icon'

const BackButton = (props) => {
  return (
    <Link
      className='back-button'
      {...props}
    >
      <Icon
        alias='arrow'
        className='back-button__icon'
      />
    </Link>
  )
}

export default BackButton
