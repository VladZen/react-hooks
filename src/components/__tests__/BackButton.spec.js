import { unmountComponentAtNode, render } from 'react-dom'
import { MemoryRouter } from 'react-router-dom'
import { act } from 'react-dom/test-utils'
import BackButton from '@/components/BackButton'

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

describe('BackButton component', () => {
  describe('rendering', () => {
    it('basic', () => {
      act(() => {
        render(
          <MemoryRouter>
            <BackButton to='/' />
          </MemoryRouter>,
          container
        )
      })

      expect(container).toMatchSnapshot()
    })

    it('don\'t render any children', () => {
      act(() => {
        render(
          <MemoryRouter>
            <BackButton to='/'>
              <a href='#'>
                Some link
              </a>
            </BackButton>
          </MemoryRouter>,
          container
        )
      })

      expect(container).toMatchSnapshot()
    })

    it('applies props to the inner link', () => {
      act(() => {
        render(
          <MemoryRouter>
            <BackButton title='Some link title' to='/' />
          </MemoryRouter>,
          container
        )
      })

      expect(container).toMatchSnapshot()
    })
  })
})
