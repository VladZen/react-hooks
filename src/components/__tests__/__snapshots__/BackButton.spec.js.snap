// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component rendering applies props to the inner link 1`] = `
<div>
  <a
    class="back-button"
    href="/"
    title="Some link title"
  >
    <svg
      class="back-button__icon"
    >
      <use
        href="sprite.svg#arrow"
      />
    </svg>
  </a>
</div>
`;

exports[`BackButton component rendering basic 1`] = `
<div>
  <a
    class="back-button"
    href="/"
  >
    <svg
      class="back-button__icon"
    >
      <use
        href="sprite.svg#arrow"
      />
    </svg>
  </a>
</div>
`;

exports[`BackButton component rendering don't render any children 1`] = `
<div>
  <a
    class="back-button"
    href="/"
  >
    <svg
      class="back-button__icon"
    >
      <use
        href="sprite.svg#arrow"
      />
    </svg>
  </a>
</div>
`;
